### DISCUSSION TITLE
Feature request: multi-map

### DISCUSSION THREAD

**Discussion by MilesCranmer** [Seelengrab/Supposition.jl#45] [Design & Ideas]
As a convenience constructor, instead of needing to use `@composed`, it would be nice to use the multiple-collection version of `map`:

```julia
julia> map((i, j) -> (i, j), Data.SampledFrom(1:10), Data.SampledFrom(5:15))
ERROR: MethodError: no method matching length(::Supposition.Data.SampledFrom{Int64, UnitRange{Int64}})
```

**Comment by Seelengrab** [comment]
Yeah, generating tuples of things is a bit of a blindspot at the moment - I'll see if I can get on this tomorrow. Shouldn't be too big of a problem, the relevant code in `src/data.jl:136-187` just needs to be adjusted to handle the multi-arg case :)

**Reply by Seelengrab** [reply]
Alright, this should be up on `feat/multimap` - let me know if that works for #47 or if there's still something missing!

**Reply by MilesCranmer** [reply]
Awesome, that works, thanks!

**Reply by Seelengrab** [reply]
Nice - if nothing disastrous shows up from your side, I'll make a release with this in the coming week.
